lib/ansible/module_utils/network/checkpoint/checkpoint.py

```diff
@@ -26,34 +26,203 @@
+        # Check the status of the task
+        code, response = send_request(connection, version, 'show-task', task_id_payload)
+
+        attempts_counter = 0
+        while code != 200:
+            if attempts_counter < 5:
+                attempts_counter += 1
+                time.sleep(2)
+                code, response = send_request(connection, version, 'show-task', task_id_payload)
+            else:
+                response['message'] = "ERROR: Failed to handle asynchronous tasks as synchronous, tasks result is" \
+                                      " undefined.\n" + response['message']
+                module.fail_json(msg=response)
+
+        # Count the number of tasks that are not in-progress
+        completed_tasks = 0
+        for task in response['tasks']:
+            if task['status'] == 'failed':
+                module.fail_json(msg='Task {0} with task id {1} failed. Look at the logs for more details'
+                                 .format(task['task-name'], task['task-id']))
+            if task['status'] == 'in progress':
+                break
+            completed_tasks += 1
+
+        # Are we done? check if all tasks are completed
+        if completed_tasks == len(response["tasks"]):
+            task_complete = True
+        else:
+            time.sleep(2)  # Wait for two seconds
+    if not task_complete:
+        module.fail_json(msg="ERROR: Timeout.\nTask-id: {0}.".format(task_id_payload['task-id']))
+
+
+# handle publish command, and wait for it to end if the user asked so
+def handle_publish(module, connection, version):
+    if module.params['auto_publish_session']:
+        publish_code, publish_response = send_request(connection, version, 'publish')
+        if publish_code != 200:
+            module.fail_json(msg=publish_response)
+        if module.params['wait_for_task']:
+            wait_for_task(module, version, connection, publish_response['task-id'])
+
+
+# handle a command
+def api_command(module, command):
+    payload = get_payload_from_parameters(module)
+    connection = Connection(module._socket_path)
+    # if user insert a specific version, we add it to the url
+    version = ('v' + module.params['version'] + '/') if module.params['version'] else ''
+
+    code, response = send_request(connection, version, command, payload)
+    result = {'changed': True}
+
+    if code == 200:
+        if module.params['wait_for_task']:
+            if 'task-id' in response:
+                wait_for_task(module, version, connection, response['task-id'])
+            elif 'tasks' in response:
+                for task_id in response['tasks']:
+                    wait_for_task(module, version, connection, task_id)
+
+        result[command] = response
+    else:
+        module.fail_json(msg='Checkpoint device returned error {0} with message {1}'.format(code, response))
+
+    return result
+
+
+# handle api call facts
+def api_call_facts(module, api_call_object, api_call_object_plural_version):
```

Comment:
@chkp-orso Is this required, if we'll be handling facts module under lookup plugins

Reply:
amm u r right, I will delete it

Reply:
ok